### PR TITLE
first abs build

### PR DESCRIPTION
**Pull request by mcg1969**
- Remove noarch
- Added abs.yaml for testing
- Updated to 2022.07.04
- Added doc_source_url, description, dev_url
- Windows fix
- Added setuptools, wheel to host
- Bumped pip version lower bound per upstream source requirements

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: 6ac0e7a8bbe02859d46c53c03d3f33136bbf49c8c7d9a6d818409e17a604eda0
+  sha256: {{ sha256 }}
 
 build:
-  noarch: python
+  # noarch: python
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:14]: Could you please add `skip: True  # [py<37]`? See https://github.com/pypa/pipenv/blob/14bc6de4c0fde7fcfc4b3945bdb1d9b61e500944/setup.py#L72

File: recipe/meta.yaml [hunk 2]
```diff
@@ -19,12 +20,14 @@ build:
 
 requirements:
   host:
-    - pip >=18.0
-    - python >=3.6
+    - pip >=22.0.4
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: `pip` should be without pinning in `host`, see https://github.com/pypa/pipenv/blob/14bc6de4c0fde7fcfc4b3945bdb1d9b61e500944/setup.py#L74

File: recipe/meta.yaml [hunk 3]
```diff
@@ -41,16 +44,23 @@ test:
     # Related:
     # https://github.com/pypa/pipenv/pull/4944
     # https://github.com/pypa/pipenv/issues/4909
-    - CI=1 python -c "import pipenv"
-    - CI=1 pipenv --help
+    - export CI=1  # [not win]
+    - set CI=1     # [win]
+    - python -c "import pipenv"
+    - pipenv --help
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:53]: It would be nice to add `pip check`

File: recipe/meta.yaml (hunk 2 above)
Comment (mcg1969) [reply recipe/meta.yaml]: Agreed. But I did end up putting it in `run` due to
https://github.com/pypa/pipenv/blob/14bc6de4c0fde7fcfc4b3945bdb1d9b61e500944/setup.py#L24
is that ok?

File: recipe/meta.yaml (hunk 3 above)
Comment (mcg1969) [reply recipe/meta.yaml:53]: done

File: recipe/meta.yaml (hunk 2 above)
Comment (mcg1969) [reply recipe/meta.yaml]: done

File: recipe/meta.yaml (hunk 1 above)
Comment (mcg1969) [reply recipe/meta.yaml:14]: done

**Comment by mcg1969** [comment]
I've incorporated the feedback and done a fresh rebuild on all platforms (deleting the packages in my test channel first).

**Review by skupr-anaconda — APPROVED** [review]

**Pull request merged**